src/components/editor/List.jsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import ReactList from 'react-list';
+import { Link } from 'react-router';
+
+export default class List extends React.Component {
+  constructor(props) {
+    super(props);
+    this.renderItem = this.renderItem.bind(this);
+  }
+  renderItem(index, key) {
+    index = this.props.indexConverter[index];
```

Comment:
Don't do this, just pass indexes in already converted. This makes a confusing API and I think its tying logic between components and creating dependencies between them.